src/views/Events/index.js

```diff
@@ -50,22 +50,108 @@ export default class Events extends Component {
     this.isMobileView = false;
     this.breakpointWidth = 540;
   }
-  componentWillMount() {
+
+  componentDidUpdate() {
+    window.onpopstate = () => {
+      if (!window.location.href.includes('?')) {
+        window.location.reload();
+      } else {
+        this.loadEvents();
+      }
+    };
+  }
+
+  componentDidMount() {
     this.loadEvents();
+    window.addEventListener('resize', this.resize);
   }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  async loadPrevious() {
+    if (window.location.href.includes('?')) {
+      const urlParams = new URLSearchParams(this.props.location.search);
+      let includePast = urlParams.get('Past') === 'true' ? true : false;
+      let chapelCredits = urlParams.get('CLW') === 'true' ? true : false;
+      let academics = urlParams.get('Academics') === 'true' ? true : false;
+      let admissions = urlParams.get('Admissions') === 'true' ? true : false;
+      let art = urlParams.get('Arts') === 'true' ? true : false;
+      let athletics = urlParams.get('Athletics') === 'true' ? true : false;
+      let calendar = urlParams.get('Calendar') === 'true' ? true : false;
+      let cec = urlParams.get('CEC') === 'true' ? true : false;
+      let fair = urlParams.get('Fair') === 'true' ? true : false;
+      let chapelOffice = urlParams.get('ChapelOffice') === 'true' ? true : false;
+      let studentLife = urlParams.get('StudentLife') === 'true' ? true : false;
+
+      this.setState({
+        includePast,
+        chapelCredits,
+        academics,
+        admissions,
+        art,
+        athletics,
+        calendar,
+        cec,
+        fair,
+        chapelOffice,
+        studentLife,
+      });
+
+      this.setState({ loading: true });
+      const events = await gordonEvent.getFilteredEvents(this.state);
+      this.setState({ filteredEvents: events, loading: false });
+
+      // If the include past filter is on, we get the events from the past
+      if (this.state.includePast) {
+        /* When togglePastEvents() runs, it changes the state of includePast to its opposite.
+         * So, when includePast is true, we set it to false so that togglePastEvents() will set it
+         * back to true
+         */
+        this.setState({ includePast: false });
+        this.togglePastEvents();
```

Comment:
Might be good to put the state change after the function call. I could see the re-rendering causing some problems because it technically re-renders before getting to the function call.

Reply:
When togglePastEvents() runs, it changes the state of includePast to its opposite. So, when includePast is true, we set it to false so that togglePastEvents() will set it back to true.

If I put it after the function call then if i had the include past filter selected and refreshed the page, it would go away.